src/models/thread.ts

```diff
@@ -504,17 +504,43 @@ export class Thread extends ReadReceipt<EmittedEvents, EventHandlerMap> {
         throw new Error("Unsupported function on the thread model");
     }
 
+    public getEventReadUpTo(userId: string, ignoreSynthesized?: boolean): string | null {
```

Comment:
It would be nice to give this a doc-comment. What does it return? what does a result of `null` mean? what does ignoreSynthesized do?

Reply:
also, uh, where is it actually used? It seems to be a new method but I can't find any new calls outside the tests?

Reply:
oh wait, `Thread` extends `ReadReceipt`? that implies a thread is a particular type read receipt? WTF does "read up to" mean for a single read receipt? I am so confused.

Reply:
I've added doc comment, documenting them similarly to what the super class does.

`ReadReceipt` is pretty poorly named I have to admit. It's an abstract class that hosts all the logic related to read receipts between a `Thread` and a `Room`.

`Thread` require a bit of extra logic as clients have gained threaded abilities somewhat recently, and it is possible for a thread to have no receipts at all